### PR TITLE
Search in sidebar experiment

### PR DESCRIPTION
**Pull request by jivey**
for: https://github.com/openstax/unified/issues/2047

Decided it was better to just copy the component into something separate and save a refactor for after we decide which placement is better.

### CODE REVIEW

**Comment by jivey** [comment]
I think I confused codecov when I rebased main and untidied the history.

**Review by Dantemss — COMMENTED** [review]
1 thing to maybe figure out:

The first time I visited your Heroku instance, the search button on the sidebar was missing and I was very confused (the normal search bar did display instead).

**Comment by Dantemss** [comment]
There's also a couple of failing tests for search.

**Review by Dantemss — APPROVED** [review]
Oh I get it, it's only a 50% chance the search will be in the sidebar. OK, it works. I'll approve, but there's still the failing test.

**Comment by jivey** [comment]
> 1 thing to maybe figure out:
> 
> The first time I visited your Heroku instance, the search button on the sidebar was missing and I was very confused (the normal search bar did display instead).

Oops I meant to document that, to see the alternate render, you have to change the end of the `_gaexp` cookie to `1` (`0` is for the existing design.) I'll check out the specs.

File: src/app/messages/es/messages.json [hunk 1]
```diff
@@ -154,10 +154,12 @@
   "i18n:prevnext:prev:aria-label": "Página anterior",
   "i18n:prevnext:prev:text": "Anterior",
   "i18n:search-results:bar": "Barra lateral de resultados de búsqueda",
-  "i18n:search-results:bar:header:title": "Resultados de búsqueda",
+  "i18n:search-results:bar:header:title:plain": "Buscar",
```
Comment (jivey) [inline src/app/messages/es/messages.json:157]: @bethshook If you get a chance, can you peek at these and let me know if they sound right? 🙏

File: src/app/messages/pl/messages.json [hunk 2]
```diff
@@ -154,10 +154,12 @@
   "i18n:prevnext:prev:aria-label": "Poprzednia strona",
   "i18n:prevnext:prev:text": "Poprzednia",
   "i18n:search-results:bar": "Wyniki wyszukiwania",
-  "i18n:search-results:bar:header:title": "Wyniki wyszukiwania",
+  "i18n:search-results:bar:header:title:plain": "Szukaj",
+  "i18n:search-results:bar:header:title:results": "Wyniki wyszukiwania",
```
Comment (jivey) [inline src/app/messages/pl/messages.json:158]: @paulinaszczucinska 👋 Is anyone be available to provide translations for these additions? I added these placeholders, but I'm unsure if they sound right.

File: src/app/messages/es/messages.json [hunk 3]
```diff
@@ -211,8 +213,11 @@
   "i18n:toolbar:practice-questions:button:text": "Práctica",
   "i18n:toolbar:print:aria-label": "Imprimir",
   "i18n:toolbar:print:text": "Imprimir",
+  "i18n:toolbar:search:text": "Buscar",
   "i18n:toolbar:search:placeholder": "Buscar en este libro",
   "i18n:toolbar:search:toggle": "Buscar en este libro",
+  "i18n:toolbar:search:toggle:opened": "Haga clic para abrir Buscar",
+  "i18n:toolbar:search:toggle:closed": "Haga clic para cerrar Buscar",
```
Comment (bethshook) [inline src/app/messages/es/messages.json]: For the two above I would change to:
`Haga clic para abrir búsqueda`
`Haca clic para cerrar búsqueda`

The rest look good.

File: src/app/messages/pl/messages.json (hunk 2 above)
Comment (jivey) [reply src/app/messages/pl/messages.json:158]: Also tagging @ewaforman in case they have some time to look

Comment (paulinaszczucinska) [reply src/app/messages/pl/messages.json:158]: Thank you @jivey! @ewaforman could you please take a look and provide these translations where necessary?

Comment (ewaforman) [reply src/app/messages/pl/messages.json:158]: @jivey @paulinaszczucinska Translations sound good.

Comment (jivey) [reply src/app/messages/pl/messages.json:158]: @ewaforman @paulinaszczucinska Thank ya'll!

File: src/app/content/search/hooks/index.ts [hunk 4]
```diff
@@ -146,10 +146,40 @@ export const syncSearch: RouteHookBody<typeof content> = (services) => async() =
...
+        ...systemQueryParams,
+      }),
+    }
+    : { search: queryString.stringify({ query }) };
+
+  services.history.replace({
+    hash,
+    ...options,
+  });
+};
```
Comment (jivey) [inline src/app/content/search/hooks/index.ts:176]: @bethshook Curious what you think about this approach for restoring the last search state?

**Comment by jivey** [comment]
@Dantemss @bethshook Just checking if ya'll had any further feedback before I move this to review

**Review by Dantemss — APPROVED** [review]

**Pull request merged**